Update kweb

fbartels opened the pull request:
fixes #208

Signed-off-by: Felix Bartels <felix@host-consultants.de>

fbartels committed:
Update kweb

fixes #208

Signed-off-by: Felix Bartels <felix@host-consultants.de>

fbartels committed:
add git to build container
improve ci testing
add kopano-ical ports to setup

Signed-off-by: Felix Bartels <felix@host-consultants.de>

fbartels committed:
fix ssl cert generation

pending further tests with lets encrypt cert generation

Signed-off-by: Felix Bartels <felix@host-consultants.de>

fbartels committed:
add todo

Signed-off-by: Felix Bartels <felix@host-consultants.de>

fbartels committed:
the fqdn needs to be part of kweb.cfg

Signed-off-by: Felix Bartels <felix@host-consultants.de>

fbartels committed:
do not need dockerize, kweb can du env replacement on its own

Signed-off-by: Felix Bartels <felix@host-consultants.de>

fbartels committed:
remove debug code

Signed-off-by: Felix Bartels <felix@host-consultants.de>

longsleep commented on web/Dockerfile:
```diff
@@ -1,11 +1,18 @@
 ENV CODE_VERSION="${CODE_VERSION}"
 
+ENV KWEBD_USER root
+ENV KWEBD_GROUP root
```
Why run kwebd as root?

fbartels replied:
with setcap in the upstream image, this is not necessary any longer

longsleep commented on web/kweb.cfg:
```diff
@@ -1,8 +1,9 @@
 
-*, :443 {
+# kweb >=0.7.0 needs set the explicit hostname (domain needs to be detectable by sni)
+{%FQDN%}, *, *:443 {
```
It should not be required to add the FQDN here expliclty.

fbartels replied:
after investigation we came to the conclusion that for lets encrypt cert generation to work the fqdn indeed needs to be included

longsleep commented on web/Dockerfile:
```diff
@@ -1,11 +1,18 @@
+ENV KWEBD_USER root
+ENV KWEBD_GROUP root
+# hadolint ignore=DL3002
+USER root
```
This should change to a non root user later on again.

fbartels replied:
without `USER root` kweb does not start since its listening on privileged ports: `2019/08/05 11:24:42 Listen: listen tcp :80: bind: permission denied
`

longsleep replied:
Just change internal ports to 8080 and 8443 should do the trick

longsleep replied:
Ah I remember now - not possible because of internal requests right?

fbartels replied:
yes, that was indeed the reason behind this.

longsleep replied:
So @fbartels to fix this set the caps on the kwebd file

`setcap 'cap_net_bind_service=+ep'  /path/to/kwebd`

longsleep replied:
Guess that is something worth adding to the standard kwebd docker image

fbartels committed:
make 2015 explicit

Signed-off-by: Felix Bartels <felix@host-consultants.de>

fbartels committed:
no longer run kweb as root

now possible because of https://stash.kopano.io/projects/KGOL/repos/kweb/commits/fd5c7307db6ebfb8e326edfa98114be7c062f77d

Signed-off-by: Felix Bartels <felix@host-consultants.de>

fbartels committed:
final changes

Signed-off-by: Felix Bartels <felix@host-consultants.de>
